packages/nouns-contracts/test/foundry/DAOUpgrade/DAOUpgradeMainnetFork.t.sol

```diff
@@ -0,0 +1,185 @@
+        vm.warp(propInfo.eta + 1);
+        NOUNS_DAO_PROXY_MAINNET.execute(proposalId);
+    }
+}
+
+contract DAOUpgradeMainnetForkTest is DAOUpgradeMainnetForkBaseTest {
+    function setUp() public override {
+        super.setUp();
+
+        // Deploy the latest DAO logic
+        vm.setEnv('DEPLOYER_PRIVATE_KEY', '0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa');
+        newLogic = address(new NounsDAOLogicV4());
+
+        // Propose the upgrade
+        vm.setEnv('PROPOSER_KEY', '0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb');
+        vm.setEnv('DAO_V3_IMPL', Strings.toHexString(uint160(newLogic), 20));
+        vm.setEnv('PROPOSAL_DESCRIPTION_FILE', 'test/foundry/DAOUpgrade/proposal-description.txt');
+        uint256 proposalId = new ProposeDAOUpgradeMainnet().run();
+
+        // Execute the upgrade
+        voteAndExecuteProposal(proposalId);
+    }
+
+    function test_daoUpgradeWorked() public {
+        assertTrue(CURRENT_DAO_IMPL != NOUNS_DAO_PROXY_MAINNET.implementation());
+        assertEq(newLogic, NOUNS_DAO_PROXY_MAINNET.implementation());
+    }
+
+    function test_givenRecentBitPacking_creationBlockAndProposalIdValuesAreLegit() public {
+        NounsDAOTypes.ProposalCondensed memory prop = NOUNS_DAO_PROXY_MAINNET.proposalsV3(493);
+
+        assertEq(prop.id, 493);
+        assertEq(prop.creationBlock, 19093670);
+        assertEq(prop.creationTimestamp, 0);
+
+        prop = NOUNS_DAO_PROXY_MAINNET.proposalsV3(474);
+
+        assertEq(prop.id, 474);
+        assertEq(prop.creationBlock, 18836862);
+        assertEq(prop.creationTimestamp, 0);
+    }
+
+    function test_adminFunctions_workUsingTheNewFallbackDesign() public {
+        uint256 currentForkPeriod = NOUNS_DAO_PROXY_MAINNET.forkPeriod();
+        uint256 expectedForkPeriod = currentForkPeriod + 1;
+
+        uint256 proposalId = propose(
+            address(NOUNS_DAO_PROXY_MAINNET),
+            0,
+            '_setForkPeriod(uint256)',
+            abi.encode(expectedForkPeriod)
+        );
+        voteAndExecuteProposal(proposalId);
+
+        assertEq(expectedForkPeriod, NOUNS_DAO_PROXY_MAINNET.forkPeriod());
+
+        uint256 currentVotingDelay = NOUNS_DAO_PROXY_MAINNET.votingDelay();
+        uint256 expectedVotingDelay = currentVotingDelay - 1;
+
+        proposalId = propose(
+            address(NOUNS_DAO_PROXY_MAINNET),
+            0,
+            '_setVotingDelay(uint256)',
+            abi.encode(expectedVotingDelay)
+        );
+        voteAndExecuteProposal(proposalId);
+
+        assertEq(expectedVotingDelay, NOUNS_DAO_PROXY_MAINNET.votingDelay());
+    }
+
+    function test_clientId_savedOnProposals() public {
+        uint32 expectedClientId = 42;
+        uint256 proposalId = propose(address(NOUNS_DAO_PROXY_MAINNET), 0, '', '', expectedClientId);
+
+        assertEq(expectedClientId, NOUNS_DAO_PROXY_MAINNET.proposalClientId(proposalId));
```

Comment:
@eladmallel you're using `proposalClientId` and `proposalVoteClientData` below which I think are not actually used by the rewards contract any more.
I thought we could delete if they aren't used. But maybe it's useful to keep them? wdyt?

Reply:
agreed and removed those functions